### DISCUSSION TITLE
FormatException: Unrecognized Guid format. - when importing solution into vanilla environment

### DISCUSSION THREAD

**Discussion by ameed** [microsoft/powerplatform-build-tools#1040] [General]
We tried to make first import of solution to a new environment using release pipeline and Az DevOps PP Build Tools and got this error even before solution import started. This hinted us that something was wrong with the solution zip file. However when tried to import the solution to an existing environment (where the solution was already earlier imported), this error did not occur.

On debugging, we found that recently something changed in solution export process, for environment variables. If an env variable is defined with a default value is set in Source Environment, the environmentvariablevalues.json file does not have the environmentvariablevalueid attribute, so the JSON looks like following. 
{
  "environmentvariablevalues": {
    "environmentvariablevalue": {
      "iscustomizable": "1"
    }
  }
}

In our source control we found this change first show up on Sept 9 as we exported using PP Build Tools, so previously the same JSON looked like this:
{
  "environmentvariablevalues": {
    "environmentvariablevalue": {
      "@environmentvariablevalueid": "953f8826-85a9-ed11-aacf-000d3a0c29b6",
      "iscustomizable": "1"
    }
  }
}

When i looked into PAC log file, i found the following stack trace, which clearly mentions that exception is arising from trying to parse a blank string as Guid. Using ILSpy, i located "GenerateSchemaNameToValueIdMapping" method in "bolt.dataverse.dll". I shall also report this issue to Microsoft, but i believe if guid of an environmentvariable value is not provide in solution zip package then import process should not try to create a value during import process 

2024-09-17 01:49:58.2572 | ERR | bolt.dataverse.client.actions.SolutionImportAction :: Caught Exception in ActionBase of: Unrecognized Guid format.|EXCEPTION: System.FormatException: Unrecognized Guid format.
   at System.Guid.GuidResult.SetFailure(ParseFailureKind failure, String failureMessageID, Object failureMessageFormatArgument, String failureArgumentName, Exception innerException)
   at System.Guid.TryParseGuid(String g, GuidStyles flags, GuidResult& result)
   at System.Guid..ctor(String g)
   at bolt.dataverse.client.actions.SolutionImportAction.GenerateSchemaNameToValueIdMapping(List`1 componentDetails)
   at bolt.dataverse.client.actions.SolutionImportAction.Execute(String solutionZipPath, ImportSolutionSettings importSettings)

**Comment by 12226317** [comment]
+follow

Same issue

annoyingly if you manually import then update you get 
"The reason given was: CanvasApp import: FAILURE: Code: GraphUserDetailNotFound Message: The user '58b92e5d-9450-4471-8922-36e0113777dd' was not found in Graph. Client Request Id: 914e0c21-11fb-4675-836a-bb0144d2a6a1."

**Comment by 568182** [comment]
**We also facing exactly the same issue, When we deploy solution into new production environment.**

**Note:** When we deploy the same solution in test environment (sandbox) below error did not occur.

[ ' info: bolt.dataverse.client.actions.SolutionImportAction[0]' ]
[ ' Executing SolutionImportAction' ]
[ ' Solution Importing...' ]
[ ' fail: bolt.dataverse.client.actions.SolutionImportAction[0]' ]
[ ' Caught Exception in ActionBase of: Unrecognized Guid format.' ]
[ ' System.FormatException: Unrecognized Guid format.' ]
[ ' at System.Guid.GuidResult.SetFailure(ParseFailureKind failure, String failureMessageID, Object failureMessageFormatArgument, String failureArgumentName, Exception innerException)']
[ ' at System.Guid.TryParseGuid(String g, GuidStyles flags, GuidResult& result)' ]
[ ' at System.Guid..ctor(String g)' ]
[ 'at bolt.dataverse.client.actions.SolutionImportAction.GenerateSchemaNameToValueIdMapping(List`1 componentDetails)' ]
[ ' at bolt.dataverse.client.actions.SolutionImportAction.Execute(String solutionZipPath, ImportSolutionSettings importSettings)']
[ 'Microsoft PowerPlatform CLI' ]
[ 'Version: 1.34.4+gbc3320d' ]
[ 'Online documentation: https://aka.ms/PowerPlatformCLI' ]
[ ' Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions' ]
[ 'Error: Unrecognized Guid format.' ]
[ 'info: bolt.cli.CommandProcessor[0]' ]
[ ' EndExecutingVerb (Success: False, Execution Time(ms): 117)' ]
[ 'info: bolt.Session[0]' ]
[ ' End session. (sessionId: b89cd516-d996-4f8e-9724-8c712171eaa1; ResponseCode: 500)']
[ ' ' ]
##[error]error: 1
##[debug]Processed: ##vso[task.issue type=error;source=TaskInternal;]error: 1
##[error]failed: info: bolt.Session[0]
Product Version: 1.34.4+gbc3320d; File Version: 1.34.4
info: bolt.Session[0]

**Comment by tocri** [comment]
same here... :('

**Comment by 12226317** [comment]
By the way, I forgot to mention—I believe I've found the solution. Check your environmental variables and set the advanced export variable to false.

*Also ensure if you are using DevOps and setting the connection references that you have shared the connection with the SPN if you have a SPN perform the import activitiy 

![image](https://github.com/user-attachments/assets/98db10bf-4bb7-4350-bfeb-1535115db89c)

**Reply by ameed** [reply]
Yes, that is how worked around and also explained while reporting the issue

**Comment by petrochuk** [comment]
thank you for reporting. This is going to be fixed in next version of PAC CLI

**Reply by 12226317** [reply]
Do we know what is creating the issue?, is it the export value for environmental variables

**Reply by petrochuk** [reply]
> Do we know what is creating the issue?, is it the export value for environmental variables

yes, it fails to parse environmentvariablevalueid from solution you are trying to import

**Comment by 568182** [comment]
We have work around solution here, 
[https://github.com/microsoft/powerplatform-build-tools/issues/1041](https://github.com/microsoft/powerplatform-build-tools/issues/1041)

**Reply by petrochuk** [reply]
we released PAC CLI 1.35 which ignores invalid GUIDs

**Reply by ameed** [reply]
Hi @petrochuk ...  thank you

I just found that PP Build Tools Extension's latest version 2.0.77 is still using pac CLI 1.34. Would you know how soon the newer version with PAC CLI 1.35 will be available on marketplace so we can update PP Build Tools in ADO Orgs ?

<img width="1045" alt="image" src="https://github.com/user-attachments/assets/8f46e74e-e9e1-4013-9783-64a3cea621f8">

**Reply by petrochuk** [reply]
We'll release updated build tools within a week.

**Reply by ameed** [reply]
Thanks Andrew

**Reply by sbalakrishnanbe** [reply]
thanks @petrochuk - please let us know once the PP Build tools are updated.

**Reply by sbalakrishnanbe** [reply]
@petrochuk - Do you have any idea when the pac CLI version will be updated in the build tools?

**Reply by petrochuk** [reply]
> @petrochuk - Do you have any idea when the pac CLI version will be updated in the build tools?

It was updated 3 weeks ago in v2.0.80

https://github.com/microsoft/powerplatform-build-tools/releases/tag/v2.0.80
https://marketplace.visualstudio.com/items?itemName=microsoft-IsvExpTools.PowerPlatform-BuildTools

**Comment by ameed** [comment]
Thank you, i will close this issue here

**Reply by sbalakrishnanbe** [reply]
Hi @petrochuk, 

We are using the Power Platform tools in the Azure DevOps pipeline to import the solution. However, the Power Platform Import Solution task version 2 uses the pac cli version 1.34, causing the solution import to fail with the "Unrecognized Guid format" error. 

Can you help me with how to update the pac cli version for the solution import tasks?

Please see the complete error log below. 

![image](https://github.com/user-attachments/assets/27791766-6079-464d-a324-6e2a7490ec7d)

**[
  "The Authentication Result: '***' authenticated successfully.,Validating connection...,Connected to... RB-CRB-UK-TEST,Default organization: RB-CRB-UK-TEST,Authentication profile created,    * UNIVERSAL                                : ***     Public ,"
]
[
  'Calling pac cli inputs: solution import --path 'C:\\azp\\agent\\_work\\23\\a\\solution.zip' --async true --force-overwrite false --publish-changes false --skip-dependency-check false --convert-to-managed true --max-async-wait-time 60 --activate-plugins true --skip-lower-version false --settings-file 'C:\\azp\\agent\\_work\\23\\s\\src\settings.json'
]
[ 'Connected as ***' ]
[ 'Connected to... RB-CRB-UK-TEST' ]
[ '' ]
[ 'Solution Importing...' ]
[ 'Microsoft PowerPlatform CLI' ]
[ 'Version: 1.34.4+gbc3320d' ]
[ 'Online documentation: https://aka.ms/PowerPlatformCLI' ]
[
  'Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions'
]
[ '' ]
[ 'Error: Unrecognized Guid format.' ]
[ '' ]
[ '' ]
##[error]error: 1
##[error]failed: Connected as *****

**Reply by ameed** [reply]
Hi, I am not sure if you have latest PAC CLI v 1.35 in PP Build Tools for DevOps. PAC team says, they have fixed it in v 1.35, check in your Azure DevOps Org settings if a new version of PP Build Tools is available and installed in your AZ DevOps Org. The way to get around this problem is to remove / deactivate env variable value from Source Environment, before running CI/CD pipeline, and use deployment config json for target environment's specific values for environment variables 

Just found that PP Build Tools Extension for AzDevOps latest version 2.0.77 is still using pac CLI 1.34. I hope build team also releases a new version soon with PAC CLI v 1.35

<img width="1045" alt="image" src="https://github.com/user-attachments/assets/9acf9e5d-e95a-4353-9c76-8b2562e9a656">

**Reply by OneAndOnlyHenry** [reply]
> Hi @petrochuk,
> 
> We are using the Power Platform tools in the Azure DevOps pipeline to import the solution. However, the Power Platform Import Solution task version 2 uses the pac cli version 1.34, causing the solution import to fail with the "Unrecognized Guid format" error.
> 
> Can you help me with how to update the pac cli version for the solution import tasks?
> 
> Please see the complete error log below.
> 
> ![image](https://private-user-images.githubusercontent.com/38248404/371752641-27791766-6079-464d-a324-6e2a7490ec7d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc1OTUyMDIsIm5iZiI6MTcyNzU5NDkwMiwicGF0aCI6Ii8zODI0ODQwNC8zNzE3NTI2NDEtMjc3OTE3NjYtNjA3OS00NjRkLWEzMjQtNmUyYTc0OTBlYzdkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI5VDA3MjgyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiOGZjZGMzNTZlN2RlOWRkNjU1MjJlZjdhNDUyNTIxMzhmNjk5OWYwM2NjZTEwYjkxNGQ5NGUxNDZlOGNmODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DpwpBCKT4WkvQ90REX8uNsFglg6ihpxtq_Bj4RzKasc)
> 
> **[ "The Authentication Result: '***' authenticated successfully.,Validating connection...,Connected to... RB-CRB-UK-TEST,Default organization: RB-CRB-UK-TEST,Authentication profile created, * UNIVERSAL : *** Public ," ] [ 'Calling pac cli inputs: solution import --path 'C:\azp\agent\_work\23\a\solution.zip' --async true --force-overwrite false --publish-changes false --skip-dependency-check false --convert-to-managed true --max-async-wait-time 60 --activate-plugins true --skip-lower-version false --settings-file 'C:\azp\agent\_work\23\s\src\settings.json' ] [ 'Connected as ***' ] [ 'Connected to... RB-CRB-UK-TEST' ] [ '' ] [ 'Solution Importing...' ] [ 'Microsoft PowerPlatform CLI' ] [ 'Version: 1.34.4+gbc3320d' ] [ 'Online documentation: https://aka.ms/PowerPlatformCLI' ] [ 'Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions' ] [ '' ] [ 'Error: Unrecognized Guid format.' ] [ '' ] [ '' ] ##[error]error: 1 ##[error]failed: Connected as *****

I was having the same problem but I don’t want to wait for the new version of power platform-actions and I managed to fix it by overriding the old version and make use of the new version 1.35.1

my script writes as below and it finally worked for me and no more “Unrecognized Guid format” error 

steps:
    - name: Install Power Platform Tools
      id: install-power-platform-tools
      uses: microsoft/powerplatform-actions/actions-install@v1
      with:
        pac-version-override: 1.35.1

hope it helps, cheers!

btw, I am using GitHub actions, so it might be different than your case since you are using azure DevOps, but there might still be ways to override it

**Reply by sbalakrishnanbe** [reply]
@OneAndOnlyHenry - Unfortunately, the Azure PP Build Tools do not have the option to override the pac version.
